packages/nimble-components/src/table/testing/table.pageobject.ts

```diff
@@ -207,6 +208,18 @@ export class TablePageObject<T extends TableRecord> {
         return iconOrSpinner.tagName.toLocaleLowerCase();
     }
 
+    public getRenderedMappingColumnCellIconSeverity(
+        rowIndex: number,
+        columnIndex: number
+    ): IconSeverity {
+        const iconOrSpinner = this.getRenderedMappingColumnIconOrSpinner(
+            this.getRenderedCellView(rowIndex, columnIndex)
+        );
+        return iconOrSpinner instanceof Icon
+            ? iconOrSpinner.severity
+            : undefined;
```

Comment:
Other page object methods generally throw in scenarios where preconditions fail (i.e. this is supposed to be used on cells that have icons and instead non-icon cells are silently ignored).

Also this approach makes it not possible to distinguish between an existing icon of severity default (which is the value undefined) and an incorrect mapping (where this method returns undefined). Think it should be consistent in strictness and not ambiguous in return value.

Reply:
The call to `getRenderedMappingColumnIconOrSpinner` throws if the cell isn't rendering an icon or spinner, but I agree with your point that we should also throw if it is a spinner.